lib/grape_entity_matchers/document_matcher.rb

```diff
@@ -0,0 +1,98 @@
+module GrapeEntityMatchers
+  module DocumentMatcher
+    def document(documentable)
+      DocumentMatcher.new(documentable)
+    end
+
+    class DocumentMatcher
```

Comment:
A `DocumentMatcher` class within a `DocumentMatcher` module is a bit confusing. Is this consistent with the rest of the gem implementation?